cmd/clusterctl/client/config.go

```diff
@@ -58,6 +58,38 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+type ProcessYAMLOptions struct {
+	// URLSource to be used for reading the template
+	URLSource *URLSourceOptions
+
+	// ListVariablesOnly return the list of variables expected by the template
+	// without executing any further processing.
+	ListVariablesOnly bool
+}
+
+func (c *clusterctlClient) ProcessYAML(options ProcessYAMLOptions) (YamlPrinter, error) {
+
+	// Technically we do not need to connect to the cluster. However, we are
+	// leveraging the template client which exposes GetFromURL() is available
+	// on the cluster client so we create a cluster client with default
+	// configs to access it.
+	cluster, err := c.clusterClientFactory(
+		ClusterClientFactoryInput{
+			// use the default kubeconfig
+			kubeconfig: Kubeconfig{},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if options.URLSource != nil {
+		return c.getTemplateFromURL(cluster, *options.URLSource, "", options.ListVariablesOnly)
+	}
+
+	return nil, errors.New("unable to read custom template. Please specify a template source")
```

Comment:
Flip these for readability? As in `options.URLSource == nil`, then return an error inside the block